### PR TITLE
added the supervised interfaces

### LINKED ISSUE #66: Suggestion suggestion: Motor subsystem implementations with supervised mode
Add some subsystems or interfaces (or options for existing things e.g. rotateable) that will allow built in guaranteed supervised behavior. e.g. don't cross certain angle or speed. (could just receive a supplier)

I can implement it if you think that it's a good idea.

### PR DESCRIPTION
**Pull request by dandeduck**
Not sure about this one... close #66

### CODE REVIEW

File: flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/motion/supervised/Supervisable.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.flash3388.flashlib.robot.motion.supervised;
+
+import com.flash3388.flashlib.control.Stoppable;
+
+public interface Supervisable extends Stoppable {
```
Comment (tomtzook) [inline flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/motion/supervised/Supervisable.java:5]: `Supervisable` isn't the best name for it. I understand where it comes from, but saying _supervision_ implies something completely different.

Maybe something around `Limit`?

File: flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/motion/supervised/SupervisedMovable.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package com.flash3388.flashlib.robot.motion.supervised;
+
+import com.flash3388.flashlib.control.Direction;
+import com.flash3388.flashlib.control.Stoppable;
+import com.flash3388.flashlib.scheduling.Requirement;
+
+public interface SupervisedMovable extends Stoppable, Supervisable, Requirement {
```
Comment (tomtzook) [inline flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/motion/supervised/SupervisedMovable.java:7]: You could just extend `Movable`... You wouldn't need to provide the defaults in that case.

Do the same with the rest.

File: flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/motion/supervised/SupervisedMovable.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    default void supervisedMove(double speed) {
```
Comment (tomtzook) [inline flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/motion/supervised/SupervisedMovable.java:9]: It kinda sounds like it should almost be implicit (the idea that the motion is _supervised_), such that a call to `move` is automatically _supervised_.

Remember that it's an interface, you don't actually need to provide all the capabilities in it with `default`, simply require someone to implement it to your specifications.

Same for rest.

File: flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/motion/supervised/SupervisedMovable2D.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import com.flash3388.flashlib.control.Direction;
+import com.flash3388.flashlib.scheduling.Requirement;
+import com.jmath.vectors.Vector2;
+
+/**
+ * Describes a component capable of moving along a 2 axes.
+ *
+ * @since FlashLib 2.0.0
+ */
+public interface SupervisedMovable2D extends SupervisedMovable, Requirement {
```
Comment (tomtzook) [inline flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/motion/supervised/SupervisedMovable2D.java:12]: You copied the documentation.... Fix it to something correct.

File: flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/motion/supervised/SupervisedMovable2D.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+     * Moves along both axes at given speeds.
+     *
+     * @param motionVector vector describing speed of motion, where
+     *                     each axis [-1...1] describes percentage of power
+     *                     and direction.
+     */
+    void move(Vector2 motionVector);
+
+    @Override
+    default void supervisedMove(double speed) {
```
Comment (tomtzook) [inline flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/motion/supervised/SupervisedMovable2D.java:33]: Why do you re-do the default here?

**Review by tomtzook — CHANGES_REQUESTED** [review]
You might want to consider providing some implementations for those interfaces.

Add documentation to your new classes and methods.

File: flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/motion/supervised/SupervisedMovable.java (hunk 3 above)
Comment (dandeduck) [reply flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/motion/supervised/SupervisedMovable.java:9]: I forgot to do it... I just copied them and didnt finish it lol

**Pull request closed without merging**